### PR TITLE
Use pyproject.toml, catch with other project developments

### PR DESCRIPTION
**Pull request by pkubanek**
- Updated tcpip version, pyproject.toml'ize
- Remove .pre-commit-conf, add ts-pre-commit-config file
- Update version-history

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,6 +1,3 @@
-[aliases]
-test=pytest
-
 [flake8]
```
Comment (wvreeven) [inline setup.cfg]: Please remove this file completely.

File: python/lsst/ts/dream/common/client.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
```
Comment (wvreeven) [inline python/lsst/ts/dream/common/client.py]: I don't see how this helps executing this file without a `__mian__` method so please remove this.

**Review by wvreeven — CHANGES_REQUESTED** [review]
I left e few minor comments for you. There are two reasons for requesting changes:
1. Please clean up the git history.
2. Please fix the CI build.

File: setup.cfg (hunk 1 above)
Comment (pkubanek) [reply setup.cfg]: That was already removed - cannot find it in my local copy.

File: python/lsst/ts/dream/common/client.py (hunk 2 above)
Comment (pkubanek) [reply python/lsst/ts/dream/common/client.py]: Removed

File: doc/version-history.rst [hunk 3]
```diff
@@ -4,6 +4,12 @@
 Version History
 ===============
 
+v0.3.1
```
Comment (wvreeven) [inline doc/version-history.rst]: Please update the version to `v0.4.0`. The change to ts_tcpip is quite large, don't you think?

**Review by wvreeven — APPROVED** [review]
Looks good to me. Please bump the version as indicated in my comment.

**Review by wvreeven — CHANGES_REQUESTED** [review]
Please add a conda recipe.

**Review by wvreeven — APPROVED** [review]
Looks good to me.

File: doc/version-history.rst (hunk 3 above)
Comment (pkubanek) [reply doc/version-history.rst]: Agree.

**Pull request merged**